lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,430 @@
+         * @param {{ messageId: string, node: ASTNode }} descriptor Report descriptor without fix or suggest
+         * @param {{ messageId: string, fix: Function }} suggestion Adds the fix or the whole suggestion as only element in "suggest" to suggestion
+         * @param {ASTNode} assignment Used to check if the fix is safe
+         * @returns {Object} Descriptor with either an added fix or suggestion
+         */
+        function createConditionalFixer(descriptor, suggestion, assignment) {
+            if (shouldBeFixed(assignment)) {
+                return {
+                    ...descriptor,
+                    fix: suggestion.fix
+                };
+            }
+
+            return {
+                ...descriptor,
+                suggest: [suggestion]
+            };
+        }
+
+
+        /**
+         * Returns the operator token for assignments and binary expressions
+         * @param {ASTNode} node AssignmentExpression or BinaryExpression
+         * @returns {import('eslint').AST.Token} Operator token between the left and right expression
+         */
+        function getOperatorToken(node) {
+            return sourceCode.getFirstTokenBetween(node.left, node.right, token => token.value === node.operator);
+        }
+
+        if (mode === "never") {
+            return {
+
+                // foo ||= bar
+                "AssignmentExpression"(assignment) {
+                    if (!astUtils.isLogicalAssignmentOperator(assignment.operator)) {
+                        return;
+                    }
+
+                    const descriptor = {
+                        messageId: "unexpected",
+                        node: assignment,
+                        data: { operator: assignment.operator }
+                    };
+                    const suggestion = {
+                        messageId: "separate",
+                        data: { operator: assignment.operator },
+                        *fix(ruleFixer) {
+                            if (sourceCode.getCommentsInside(assignment).length > 0) {
+                                return;
+                            }
+
+                            const operatorToken = getOperatorToken(assignment);
+
+                            // -> foo = bar
+                            yield ruleFixer.replaceText(operatorToken, "=");
+
+                            const assignmentText = sourceCode.getText(assignment.left);
+                            const operator = assignment.operator.slice(0, -1);
+
+                            // -> foo = foo || bar
+                            yield ruleFixer.insertTextAfter(operatorToken, ` ${assignmentText} ${operator}`);
+
+                            if (
+                                astUtils.getPrecedence(assignment.right) <= astUtils.getPrecedence({ type: "LogicalExpression", operator }) &&
+                                !astUtils.isParenthesised(sourceCode, assignment.right)
+                            ) {
```

Comment:
There's also a special case: `??` cannot be mixed with `||` or `&&` without parentheses.

I think that this check already covers all combinations, except for this one:

```js
/* eslint logical-assignment-operators: ["error", "never"] */

a ??= b && c;
``` 

After the fix:

```js
/* eslint logical-assignment-operators: ["error", "never"] */

a = a ?? b && c; // parsing error

```